### ISSUE TITLE
Save to Server instead of download to Client?

### ISSUE THREAD

**Issue by dbig** [waveform80/compoundpi#50]
Enjoying Compound Pi, thanks for providing it to the community!

I appreciate that the genesis of Compound Pi (it seems) is simultaneous image capture and presentation, for which downloading to the Client makes total sense.

My application is video-based, with files likely larger than the available ram on each Server, so I'd like to save the files on each Server's local SD card, instead of having to download it to the Client.

Can this be done?  Is it possible to create an option for this in each Server's configuration?

Thanks very much,
Doug

**Comment by waveform80** [comment]
Yup - this is planned as part of #44. However, it's going to require a fair bit of work. As part of good practice, the camera daemon drops privileges after it starts which'll restrict the places it can write to so the configuration will need to include an appropriate user to drop to (e.g. pi) and an output location that's definitely accessible. Then the net protocol needs extending to handle more file operations (and filenames!)

Still, what I think you're really asking for is an option to ensure that the daemon can leave any files it creates (I was intending just to use the SD card as temporary storage, i.e. clean up the created files at daemon shutdown, or use an entirely temporary directory and leave the OS to handle clean-up). I could certainly add that as an option. I'm guessing to be really useful, the files would need some form of sensible naming as well (again, I was planning something less useful like sequential numbering or random hashes) - i.e. before capture / record the client ought to be able to specify some string to include in the filenames, which should then be annotated with some unique server ID (e.g. IP address) and timestamp.

I'll mark this as an enhancement - can't promise this'll be turning up any time soon as to make it properly useful requires a fair amount of work, but it's on the list!

**Comment by davidhbrann** [comment]
Hi, I have the same issue. I was hoping to use Compound Pi to simultaneous record video on multiple PIs. However, given the 1GB RAM on the Pi 3, I think I'll run out of RAM fairly quickly. Do you have suggestions for how to avoid this issue when recording in batch mode? How quick can you download to the client and clear the previous video before starting again? Or has there been any progress on saving to the SD card?

If you think Compound Pi wouldn't work, do you have any alternative suggestions? If I'm less concerned about synchronization, should I just try something like parallel ssh with picamera?

Thanks for all your work on this project!
-David
